drivers/serial/uart_nrfx_uarte.c

```diff
@@ -286,8 +820,24 @@ static int uarte_nrfx_poll_in(struct device *dev, unsigned char *c)
 static void uarte_nrfx_poll_out(struct device *dev,
```

Comment:
What about `uarte_nrfx_poll_in`? Currently it will always return `-1` when `CONFIG_UART_ASYNC_API` is enabled.

Reply:
There is no possibility to handle poll_in while using asynchronous API, I will change it to return -EBUSY